### PR TITLE
Fix emscripten environment (#172)

### PR DESCRIPTION
**Pull request by syrusakbary**
Fix emscripten environment

### CODE REVIEW

File: lib/emscripten/src/memory.rs [hunk 1]
```diff
@@ -33,8 +47,8 @@ pub fn enlarge_memory(_ctx: &mut Ctx) -> u32 {
 }
 
 /// emscripten: abortOnCannotGrowMemory
-pub fn abort_on_cannot_grow_memory(ctx: &mut Ctx) -> u32 {
-    debug!("emscripten::abort_on_cannot_grow_memory");
+pub fn abort_on_cannot_grow_memory(requested_size: u32, ctx: &mut Ctx) -> u32 {
```
Comment (syrusakbary) [inline lib/emscripten/src/memory.rs]: Can we have a `abort_on_cannot_grow_memory` and `abort_on_cannot_grow_memory_old`? (with the previous signature)
We will use one or the other depending on the expected signature from the function in the module.

That way we will have backwards compatibility with old-generated emscripten files.

## How to do it?

Creating a flag in the EmscriptenGlobals storing if the signature is one or the other:
https://github.com/wasmerio/wasmer/blob/c3da14d95896c27ed492c7461bb0ab856d9778e4/lib/emscripten/src/lib.rs#L242

And then use one function or the other depending on that flag.

Comment (davidMcneil) [reply lib/emscripten/src/memory.rs]: Thanks, an issue I think we are going to run into is that the types of functions are not going to be the same and these functions get converted into the `Func` type which will be generic over different argument list types.

Comment (davidMcneil) [reply lib/emscripten/src/memory.rs]: It looks like I can potential use the `new_from_ptr` call. Is this the best way to go?

Comment (davidMcneil) [reply lib/emscripten/src/memory.rs]: I think I have a solution using the `IsExport` trait object.

**Comment by davidMcneil** [comment]
This should be ready to be merged. I tested locally with emcc version 1.38.21 and 1.38.26. The wasm files are built with version 1.38.26. Does it make sense to update these files or should I have kept the older version?

**Comment by syrusakbary** [comment]
Looks great! Merging

**Pull request merged**